docs/developers/sbt.md

```diff
@@ -279,3 +279,43 @@ Similarly, if your objective involves compiling and packaging within an intranet
 ```
 
 For more details on sbt repository configuration, please refer to the [SBT documentation](https://www.scala-sbt.org/1.x/docs/Proxy-Repositories.html).
+
+## Publish
+
+SBT supports publishing shade clients (Spark/Flink/MapReduce) to an internal Maven private repository, such as [Sonatype Nexus](https://www.sonatype.com/) or [JFrog](https://jfrog.com/help/r/jfrog-artifactory-documentation/maven-repository).
+
+Before executing the publish command, ensure that the following environment variables are correctly set:
+
+| Environment Variable   | Description                                                                                                                      |
+| ---------------------- | -------------------------------------------------------------------------------------------------------------------------------- |
+| SONATYPE_HOST          | Sonatype repository host address, default is "oss.sonatype.org"                                                                  |
+| SONATYPE_USERNAME      | Sonatype repository username                                                                                                     |
+| SONATYPE_PASSWORD      | Sonatype repository password                                                                                                     |
+| SONATYPE_SNAPSHOTS_URL | Sonatype repository URL for snapshot version releases, default is "https://oss.sonatype.org/content/repositories/snapshots"      |
+| SONATYPE_RELEASES_URL  | Sonatype repository URL for official release versions, default is "https://oss.sonatype.org/service/local/staging/deploy/maven2" |
```

Comment:
Are SONATYPE_HOST redundant? I suppose SONATYPE_RELEASES_URL and SONATYPE_SNAPSHOTS_URL are sufficient.

For Apache release, the default URL should be https://repository.apache.org/service/local/staging/deploy/maven2 and https://repository.apache.org/content/repositories/snapshots

Reply:
we can eliminate SONATYPE_USERNAME, SONATYPE_PASSWORD and simply pick up ASF_USERNAME, ASF_PASSWORD, as they were used in source/binary release process

Reply:
> For Apache release, the default URL should be https://repository.apache.org/service/local/staging/deploy/maven2 and https://repository.apache.org/content/repositories/snapshots

do you mean the default values(SONATYPE_SNAPSHOTS_URL/SONATYPE_RELEASES_URL) should be Apache URLs?

Reply:
Yea, we did such thing in another apache projects

Reply:
> we can eliminate SONATYPE_USERNAME, SONATYPE_PASSWORD and simply pick up ASF_USERNAME, ASF_PASSWORD, as they were used in source/binary release process

done